### PR TITLE
tob-worm-4-panics: using options rather than unwrap & tob-worm-9: cosmwasm test build

### PR DESCRIPTION
**Pull request by kii-dot**
4. Use of panics on invalid inputs
Description In several places, the code panics when an arithmetic overflow or underflow occurs. Panics should be reserved for programmer errors (e.g., assertion violations). Panicking on user errors dilutes the utility of the panic operation. An example appears in figure 4.1. Thecomplete_nativefunction useschecked_subto verify that the transfer includes sufficient funds to cover the fee. However, complete_nativepanics if there are insufficient funds.Thus, the function could panic on either a programmer error or a user error.

Also added 9. cw20-wrapped tests do not build
**for tob-9**
Removed the arguments and ran cargo test. Works seamlessly

### CODE REVIEW

File: cosmwasm/contracts/token-bridge/src/contract.rs
Comment (jynnantonix) [inline cosmwasm/contracts/token-bridge/src/contract.rs:878]: style nit: I think `ok_or_else` is the more idiomatic way to do this:

```suggestion
    amount = amount
        .checked_sub(fee)
        .ok_or_else(|| StdError::generic_err("Insufficient funds"))?;
```

same for the other places where we do this.

File: cosmwasm/contracts/token-bridge/src/contract.rs
Comment (jynnantonix) [reply cosmwasm/contracts/token-bridge/src/contract.rs:1228]: I think we probably don't need checked math for div and rem operations because those can only fail when the right hand side is 0, which is never the case in the code so far.  We can still leave them in case we change the multiplier in the future.  If we want to make this more ergonomic we can do something like:

```rust
    fee = fee
        .u128()
        .checked_rem(multiplier)
        .and_then(|rem| fee.u128().checked_sub(rem))
        .map(Uint128::new)
        .ok_or_else(|| StdError::generic_err("Invalid fee"))?;
```

**Review by jynnantonix — COMMENTED** [review]
Overall looks good to me but it does overlap a bit with  #1668.  I just have some style related suggestions.

File: solana/migration/src/api/add_liquidity.rs
Comment (jynnantonix) [inline solana/migration/src/api/add_liquidity.rs:100]: The return type for `ok_or_else` should be just the error type, not `Result<T, E>`.  Also there's no need for lazy evaluation here so we can use `ok_or` instead:

```suggestion
            .ok_or(SolitaireError::InsufficientFunds)?
```

**Review by jynnantonix — COMMENTED** [review]
Please fix the compile errors as well as the formatting errors in the solana code.  You can test locally by running `make fmt`, `make test`, and `make clippy` in the solana directory.

**Review by jynnantonix — APPROVED** [review]

**Review by jynnantonix — DISMISSED** [review]
Not sure if github will let you merge or if a committer needs to do it but please squash into a single commit when merging (instead of rebasing).

**Comment by jynnantonix** [comment]
Ack.  Though we need one more approval before we can merge.

@tbjump @a5-pickle @kev1n-peters

**Review by hendrikhofstadt — APPROVED** [review]

**Comment by hendrikhofstadt** [comment]
Great one!

**Comment by evan-gray** [comment]
rebasing because tests were stalled and there was a ci test fix

**Review by evan-gray — APPROVED** [review]
not sure why @jynnantonix's review got dismissed on rebase through GH, but @hendrikhofstadt's didn't 🤔

**Pull request merged**
